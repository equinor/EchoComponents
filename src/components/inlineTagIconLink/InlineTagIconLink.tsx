import { Button, Typography } from '@equinor/eds-core-react';
import React from 'react';
import { getIcon } from '../../helpers/getIcon';
import styles from './inlineTagIconLink.module.css';

export interface InlineTagIconLinkProps {
    onClickHandler?: () => void;
    tagNo: string;
    description: string;
    tagCategoryDescription?: string;
    legendColor: string;
}

export const InlineTagIconLink: React.FC<InlineTagIconLinkProps> = ({
    onClickHandler,
    tagNo,
    description,
    tagCategoryDescription,
    legendColor
}: InlineTagIconLinkProps): JSX.Element => {
    return (
        <Button variant={'ghost'} onClick={onClickHandler} className={styles.tagButton}>
            <div style={{ background: legendColor }} className={styles.icon}>
                {getIcon(tagCategoryDescription ?? '')}
            </div>
            <div>
                <Typography variant="body_short_link" className={styles.tag}>
                    {tagNo}
                </Typography>
                <Typography className={styles.description} variant="body_short">
                    {description}
                </Typography>
            </div>
        </Button>
    );
};

export default InlineTagIconLink;
