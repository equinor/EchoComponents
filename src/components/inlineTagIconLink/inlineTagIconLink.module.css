.tagButton {
    cursor: pointer;
    height: auto;
}

.description {
    display: inline-block;
    width: 100%;
    overflow: hidden;
    text-overflow: ellipsis;
    word-break: break-word;
    font-size: 10px !important;
    color: var(--echoText) !important;
    font-weight: 500 !important;
}

.tag {
    word-break: break-all;
}

.icon {
    position: relative;
    transform-origin: left top;
    width: 40px;
    height: 40px;
    border-radius: 10rem;
    padding: 8px;
    box-sizing: border-box;
    margin-right: var(--small);
}
