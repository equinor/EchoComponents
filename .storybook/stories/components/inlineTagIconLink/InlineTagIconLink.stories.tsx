import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';
import InlineTagIconLink, {
    InlineTagIconLinkProps
} from '../../../../src/components/inlineTagIconLink/InlineTagIconLink';

export default {
    title: 'Components/InlineTagIconLink',
    component: InlineTagIconLink,
    argTypes: {}
} as Meta;

const Template: Story<InlineTagIconLinkProps> = (args) => <InlineTagIconLink {...args} />;

export const Default = Template.bind({});

Default.args = {
    tagNo: 'D-VG10-0130',
    description: 'GATE VALVE',
    tagCategoryDescription: 'testset32323',
    legendColor: '#007079',
    onClickHandler: () => {
        console.log('ff');
    }
};
