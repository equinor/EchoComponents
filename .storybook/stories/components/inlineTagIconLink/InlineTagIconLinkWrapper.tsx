import { Table } from '@equinor/eds-core-react';
import React from 'react';
import InlineTagIconLink from '../../../../src/components/inlineTagIconLink/InlineTagIconLink';

export interface InlineTagIconLinkWrapperProps {
    onClickHandler?: () => void;
    tagNo: string;
    description: string;
    tagCategoryDescription?: string;
    legendColor: string;
    width?: number;
}

const { Row, Cell, Head, Body } = Table;

const InlineTagIconLinkWrapper: React.FC<InlineTagIconLinkWrapperProps> = ({
    onClickHandler,
    tagNo,
    description,
    tagCategoryDescription,
    legendColor,
    width
}: InlineTagIconLinkWrapperProps) => {
    return (
        <>
            <h2>Table</h2>
            <Table style={{ maxWidth: '600px' }}>
                <Head>
                    <Row>
                        <Cell>Text</Cell>
                        <Cell>Tag</Cell>
                        <Cell>More text</Cell>
                    </Row>
                </Head>
                <Body>
                    <Row>
                        <Cell>Lorem ipsum dolor sit amet consecteur dit lot.</Cell>
                        <Cell>
                            <InlineTagIconLink
                                onClickHandler={onClickHandler}
                                tagNo={tagNo}
                                description={description}
                                tagCategoryDescription={tagCategoryDescription}
                                legendColor={legendColor}
                            />
                        </Cell>
                        <Cell>Lorem ipsum dolor sit amet consecteur dit lot.</Cell>
                    </Row>
                    <Row>
                        <Cell>Lorem ipsum dolor sit amet consecteur dit lot.</Cell>
                        <Cell>
                            <InlineTagIconLink
                                onClickHandler={onClickHandler}
                                tagNo={'11-ELECTRICAL-FIELD-EQ'}
                                description={'TEST SEPARATOR WATER SUCTION'}
                                tagCategoryDescription={tagCategoryDescription}
                                legendColor={legendColor}
                            />
                        </Cell>
                        <Cell>Lorem ipsum dolor sit amet consecteur dit lot.</Cell>
                    </Row>
                </Body>
            </Table>
            <h2>Full width</h2>
            <p>
                <InlineTagIconLink
                    onClickHandler={onClickHandler}
                    tagNo={tagNo}
                    description={description}
                    tagCategoryDescription={tagCategoryDescription}
                    legendColor={legendColor}
                />
            </p>
        </>
    );
};

export default InlineTagIconLinkWrapper;
