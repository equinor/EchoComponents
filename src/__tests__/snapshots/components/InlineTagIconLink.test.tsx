import React from 'react';
import renderer from 'react-test-renderer';
import { InlineTagIconLink } from '../../../components/inlineTagIconLink/InlineTagIconLink';

it('renders inlineTagIconLink correctly', () => {
    const tree = renderer
        .create(
            <InlineTagIconLink
                tagNo={'test'}
                description={'221312'}
                tagCategoryDescription={'testset32323'}
                legendColor={'#FF0000'}
                onClickHandler={() => {
                    console.log('ff');
                }}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
