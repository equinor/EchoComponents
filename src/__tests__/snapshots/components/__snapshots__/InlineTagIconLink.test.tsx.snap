// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders inlineTagIconLink correctly 1`] = `
<button
  className="sc-hKgILt bDwaoy tagButton"
  disabled={false}
  onClick={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="sc-dlfnbm iOlYhO"
  >
    <div
      className="icon"
      style={
        Object {
          "background": "#FF0000",
        }
      }
    >
      <svg
        aria-label="tag"
      />
    </div>
    <div>
      <p
        className="sc-eCssSg bvfwKK tag"
      >
        test
      </p>
      <p
        className="sc-eCssSg bHjGEK description"
      >
        221312
      </p>
    </div>
  </span>
</button>
`;
